fastai/dataset.py

```diff
@@ -40,12 +40,18 @@ def read_dir(path, folder):
     return [os.path.relpath(f,path) for f in fnames]
 
 def read_dirs(path, folder):
-    full_path = os.path.join(path, folder)
-    all_labels = sorted([os.path.basename(os.path.dirname(f))
-                  for f in iglob(f"{full_path}/*/")])
-    fnames = [iglob(f"{full_path}/{d}/*.*") for d in all_labels]
-    pairs = [(os.path.relpath(fn,path), l) for l,f in zip(all_labels, fnames) for fn in f]
-    return list(zip(*pairs))+[all_labels]
+    labels = []
+    filenames = []
+    all_labels = []
+    pjoin = os.path.join
+    full_path = pjoin(path, folder)
+    os.chdir(full_path)
```

Comment:
Change Directories in the middle of code may move the current working directory without explicit consent of the user of the API

Reply:
I realised that and put in a new version without these environment changes.